src/main/java/baritone/utils/ToolSet.java

```diff
@@ -23,58 +23,109 @@
 import net.minecraft.enchantment.EnchantmentHelper;
 import net.minecraft.init.Enchantments;
 import net.minecraft.init.MobEffects;
+import net.minecraft.item.Item.ToolMaterial;
 import net.minecraft.item.ItemStack;
+import net.minecraft.item.ItemTool;
 
 import java.util.HashMap;
 import java.util.Map;
+import java.util.function.Function;
 
 /**
  * A cached list of the best tools on the hotbar for any block
  *
- * @author avecowa, Brady, leijurv
+ * @author Avery, Brady, leijurv
  */
 public class ToolSet implements Helper {
-
     /**
      * A cache mapping a {@link Block} to how long it will take to break
      * with this toolset, given the optimum tool is used.
      */
-    private Map<Block, Double> breakStrengthCache = new HashMap<>();
+    private final Map<Block, Double> breakStrengthCache;
+
+    /**
+     * My buddy leijurv owned me so we have this to not create a new lambda instance.
+     */
+    private final Function<Block, Double> backendCalculation;
+
+    public ToolSet() {
+        super();
+        breakStrengthCache = new HashMap<>();
+
+        if (Baritone.settings().considerPotionEffects.get()) {
```

Comment:
This is not worth it, returning a double gets boxed into a Double to pass through this function, we can just have an if like this to apply the potion modifier or not, no need to do this fancy business with composing the amplifier. It is quite functional though.

Reply:
The reason is this happens once ever. Otherwise on every block type you have to check the potion status.

Reply:
It also means you get to skip one or two condition jumps on every block strength calculation.

Reply:
I'm not sure what you're asking me to do... This is the final part of your review.

Reply:
Ah sorry, should have been more clear. 
I'd prefer if you had a function in this class, like `fullCalculation` or something idk that was just

```
private void fullCalculation(Block block) {  
  double ret = getBestDestructionTime(block);
  if (Baritone.settings().considerPotionEffects.get()) {
    ret *= potionAmplifier();
  }
  return ret;
}
```

And then instead of these shenannigans in the constructor, getStrVsBlock could just be `return breakStrengthCache.computeIfAbsent(state.getBlock(), this::fullCalculation);`